renku_pulumi/__main__.py

```diff
@@ -0,0 +1,159 @@
+        "jena": {},
+    },
+    "renku": {},
+    "useHTTPS": True,
+}
+
+
+def get_global_values(config):
+    global_values = pulumi.Config("global")
+    global_values = global_values.get_object("values") or {}
+
+    stack = pulumi.get_stack()
+
+    if config.get_bool("dev"):
+        # set values for dev deployment
+        default_global_values["gateway"]["clientSecret"] = RandomId(
+            "gateway_client_secret", byte_length=32
+        ).hex
+
+        default_global_values["gateway"]["gitlabClientSecret"] = RandomId(
+            "gateway_gitlab_client_secret", byte_length=32
+        ).hex
+
+        baseurl = config.get("baseurl")
+        k8s_config = pulumi.Config("kubernetes")
+
+        if baseurl:
+            default_global_values["renku"]["domain"] = "{}.{}".format(
+                k8s_config.require("namespace"), baseurl
+            )
+
+    default_global_values["graph"]["jena"]["dataset"] = stack
+
+    global_values = always_merger.merge(default_global_values, global_values)
+
+    return global_values
+
+
+def deploy():
+    config = pulumi.Config()
+
+    chart_reqs = ConfigParser()
+    chart_reqs.read("chart_requirements.ini")
+
+    global_values = get_global_values(config)
+
+    # global config is used to propagate dynamic values
+    global_config = {}
+
+    global_config["http"] = "https" if global_values["useHTTPS"] else "http"
+    global_config["global"] = global_values
+
+    if config.require_bool("gitlab_enabled"):
+        gps, gss = gitlab_secrets()
+        g = gitlab(config, global_config, chart_reqs)
+        gitlab_postinstall_job(global_config, gps, [g, gss])
+
+    renku_secret(global_config)
+
+    # ui
+    ui = renku_ui(config, global_config, chart_reqs)
+
+    cfg_map = configmap(global_config)
+
+    p = None
+    pg_job = None
+    graph = None
+    kc_job = None
+
+    pg, tok = graph_secrets()
+
+    if config.require_bool("postgres_enabled"):
+        p = postgresql(config, global_config, chart_reqs)
+
+    if config.require_bool("graph_enabled"):
+        graph = renku_graph(config, global_config, chart_reqs, pg, tok, p)
+
+    pg_job = postgres_postinstall_job(global_config, pg, tok, cfg_map, [p])
+
+    if config.require_bool("keycloak_enabled"):
+        kp, ks, kus = keycloak_secrets(global_config)
+        keycloak_chart = keycloak(config, global_config, chart_reqs, p, [pg_job])
+
+    if config.require_bool("minio_enabled"):
+        minio(config, global_config)
+
+    gateway_values = gateway.gateway_values(global_config)
+
+    js = jupyterhub_secrets()
+    nb = renku_notebooks(
+        config, global_config, chart_reqs, js, p, dependencies=[pg_job]
+    )
+
+    # gateway
+    redis_chart = redis(global_config, chart_reqs)
+    sc = gateway.secret(global_config, gateway_values)
+    cfg = gateway.configmaps(global_config, gateway_values)
+    gateway.ingress(global_config, gateway_values)
+    gd = gateway.gateway_deployment(global_config, gateway_values, sc, cfg)
```

Comment:
can we make the naming here more verbose and less cryptic?  As someone coming to understand how the platform is deployed and what pieces there are, this is probably the most critical starting point and we should make it as clear as possible.